fs/xfstests.py

```diff
@@ -444,36 +441,30 @@ def setUp(self):
     def test(self):
         failures = False
         os.chdir(self.teststmpdir)
-        if not self.test_list:
-            self.log.info('Running all tests')
-            args = ''
-            if self.exclude or self.gen_exclude:
-                args = ' -E %s' % self.exclude_file
-            cmd = './check %s -g %s' % (args, self.group)
+        if self.args:
+            cmd = f"./check {self.args}"
             result = process.run(cmd, ignore_status=True, verbose=True)
             if result.exit_status == 0:
-                self.log.info('OK: All Tests passed.')
+                self.log.info("OK: All tests passed")
             else:
                 msg = self._parse_error_message(result.stdout)
-                self.log.info('ERR: Test(s) failed. Message: %s', msg)
-                failures = True
-
-        else:
-            self.log.info('Running only specified tests')
-            for test in self.test_list:
-                test = '%s/%s' % (self.fs_to_test, test)
-                cmd = './check %s' % test
-                result = process.run(cmd, ignore_status=True, verbose=True)
-                if result.exit_status == 0:
-                    self.log.info('OK: Test %s passed.', test)
-                else:
-                    msg = self._parse_error_message(result.stdout)
-                    self.log.info('ERR: %s failed. Message: %s', test, msg)
-                    failures = True
-        if failures:
-            self.fail('One or more tests failed. Please check the logs.')
+                self.log.info("FAIL: Test(s) failed %s" % msg)
+                self.fail('One or more tests failed. Please check the logs.')
 
     def tearDown(self):
+
+        srcdir = f"{self.teststmpdir}/results"
+
+        job_dir = os.path.dirname(os.path.dirname(self.logdir))
+        self.job_id = os.path.basename(job_dir)
+
+        self.log.debug(" Job ID: %s, logdir: %s, srcdir: %s, outputdir: %s: " %
+                       (self.job_id, self.logdir, srcdir, self.outputdir))
+
+        new_outputdir = os.path.join(self.outputdir, os.path.basename(srcdir))
+        if (os.path.exists(self.outputdir)):
```

Comment:
this condition fails when src results dir is not present. eg- running btrfs on rhel
we should check if srcdir is present or not before copying
`if os.path.exists(srcdir):`

Reply:
when we run btrfs on rhel, it should cancel the test and run teardown. But in teardown since results dir is not present it throws error.
```
 (5/6) xfstests.py:Xfstests.test;run-fs_type-fs_btrfs_4k-loop_type-b9dc: ERROR: [Errno 2] No such file or directory: '/var/tmp/avocado_em29zjl5/results' (0.75 s)
 (6/6) xfstests.py:Xfstests.test;run-fs_type-fs_btrfs_64k-loop_type-ab81: ERROR: [Errno 2] No such file or directory: '/var/tmp/avocado_em29zjl5/results' (0.75 s)
```

Reply:
ok got it.. So this happens during error scenario.. I will fix it in the next rev.